src/anal_ghidra.cpp

```diff
@@ -0,0 +1,2405 @@
+				anal_op->fail = ins.getFallThrough().getOffset();
+				anal_op->eob = true;
+				break;
+
+			case FlowType::JUMP_TERMINATOR: anal_op->eob = true;
+			case FlowType::UNCONDITIONAL_JUMP:
+				anal_op->type = R_ANAL_OP_TYPE_JMP;
+				anal_op->jump = ins.getFlows().begin()->getOffset();
+				break;
+
+			case FlowType::COMPUTED_JUMP:
+			{
+				char *reg = getIndirectReg(ins, isRefed);
+				if(reg)
+				{
+					if(isRefed)
+					{
+						anal_op->type = R_ANAL_OP_TYPE_MJMP;
+						anal_op->ireg = reg;
+					}
+					else
+					{
+						anal_op->type = R_ANAL_OP_TYPE_IRJMP;
+						anal_op->reg = reg;
+					}
+				}
+				else
+					anal_op->type = R_ANAL_OP_TYPE_IJMP;
+				break;
+			}
+
+			case FlowType::CONDITIONAL_COMPUTED_JUMP:
+			{
+				char *reg = getIndirectReg(ins, isRefed);
+				if(reg)
+				{
+					if(isRefed)
+					{
+						anal_op->type = R_ANAL_OP_TYPE_MCJMP;
+						anal_op->ireg = reg;
+					}
+					else
+					{
+						anal_op->type = R_ANAL_OP_TYPE_RCJMP;
+						anal_op->reg = reg;
+					}
+				}
+				else
+					anal_op->type = R_ANAL_OP_TYPE_UCJMP;
+				anal_op->fail = ins.getFallThrough().getOffset();
+				break;
+			}
+
+			case FlowType::CONDITIONAL_JUMP:
+				anal_op->type = R_ANAL_OP_TYPE_CJMP;
+				anal_op->jump = ins.getFlows().begin()->getOffset();
+				anal_op->fail = ins.getFallThrough().getOffset();
+				break;
+
+			case FlowType::CALL_TERMINATOR: anal_op->eob = true;
+			case FlowType::UNCONDITIONAL_CALL:
+				anal_op->type = R_ANAL_OP_TYPE_CALL;
+				anal_op->jump = ins.getFlows().begin()->getOffset();
+				anal_op->fail = ins.getFallThrough().getOffset();
+				break;
+
+			case FlowType::CONDITIONAL_COMPUTED_CALL:
+			{
+				char *reg = getIndirectReg(ins, isRefed);
+				if(reg)
+					if(isRefed)
+						anal_op->ireg = reg;
+					else
+						anal_op->reg = reg;
+
+				anal_op->type = R_ANAL_OP_TYPE_UCCALL;
+				anal_op->fail = ins.getFallThrough().getOffset();
+				break;
+			}
+
+			case FlowType::CONDITIONAL_CALL:
+				anal_op->type |= R_ANAL_OP_TYPE_CCALL;
+				anal_op->jump = ins.getFlows().begin()->getOffset();
+				anal_op->fail = ins.getFallThrough().getOffset();
+				break;
+
+			case FlowType::COMPUTED_CALL_TERMINATOR: anal_op->eob = true;
+			case FlowType::COMPUTED_CALL:
+			{
+				char *reg = getIndirectReg(ins, isRefed);
+				if(reg)
+				{
+					if(isRefed)
+					{
+						anal_op->type = R_ANAL_OP_TYPE_IRCALL;
+						anal_op->ireg = reg;
+					}
+					else
+					{
+						anal_op->type = R_ANAL_OP_TYPE_IRCALL;
+						anal_op->reg = reg;
+					}
+				}
+				else
+					anal_op->type = R_ANAL_OP_TYPE_ICALL;
+				anal_op->fail = ins.getFallThrough().getOffset();
+				break;
+			}
+
+			default: throw LowlevelError("Unexpected FlowType occured in sleigh_op.");
+		}
+	}
+	else // Label each instruction based on a series of P-codes.
+		anal_type(a, anal_op, pcode_slg, assem);
+
+	return anal_op->size;
+}
+
+static char *get_reg_profile(RAnal *anal)
+{
+	if(!strcmp(anal->cpu, "x86"))
+		return nullptr;
```

Comment:
What's the reason for this?

Reply:
To avoid unnecessary initialization when radare just try to guess what arch current binary is.

Reply:
So "x86" in this case is because it's the default arch that r2 sets before knowing it from the binary? This could be problematic because the default arch in r2 is dependent on the host architecture, i.e. if you work on a raspberry pi or something, it would be "arm".
In any case, I think this needs at least a comment so it is clear why this check is there.

Reply:
Yes please add a comment.